src/plugins/vis_type_vega/public/vega_request_handler.ts

```diff
@@ -53,6 +50,10 @@ export function createVegaRequestHandler({
 
     const esQueryConfigs = esQuery.getEsQueryConfig(uiSettings);
     const filtersDsl = esQuery.buildEsQuery(undefined, query, filters, esQueryConfigs);
+    const searchCache = new SearchCache(getData().search.__LEGACY.esClient, {
```

Comment:
Please store the search cache instance somewhere outside of the handler function so it's reusable between requests. Something like
```ts
let searchCache: SearchCache | undefined = undefined;
return ({ timeRange, filters, query, visParams }: VegaRequestHandlerParams) => {
  if (!searchCache) {
    searchCache = new SearchCache(getData().search.__LEGACY.esClient, {
      max: 10,
      maxAge: 4 * 1000,
    });
  }
//...
```